src/CloudVariable/Encoder.php

```diff
@@ -0,0 +1,20 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\MagentoCloud\CloudVariable;
```

Comment:
This namespace is not correct. You call namespace of purpose instead of calling implementation for purpose.
It should be:
```
Magetno\MagentoCloud\Encoder
-- EnvironemntVariableEncoder
-- EncoderInterface
```

Reply:
DecoderInterface that I added is resposible only for decoding cloud variable It shouldn't be in general `Encoder` space because it has specific purpose. It should be in some `CloudVariable` or just `Variable` space.

Reply:
your interface `DecoderInterface` does not belong to space `CloudVariable` -  it is a generic interface which supposes to be used for more decoding implementations.

Reply:
It's not a generic Decoder Interface it's a specific interface for decoding cloud variables. It has only one purpose, all implementation should be used only for decoding cloud variables.

Reply:
This does not make much sense, but in this case, rename namespace to `PlatformVariable`

Reply:
Done